GeometryCore/src/Mesh.cpp

```diff
@@ -208,4 +210,52 @@ namespace geom_utils
 		}
 		return foundation;
 	}
+	Mesh integrateFoundationIntoModel(const Mesh& model,const Mesh& foundation)
+	{
+		Mesh resultFigure;
+		Mesh copyModel(model);
```

Comment:
For translate method,i need copyFoundation,or no?

Reply:
For translate() - yes. Because translate is non-const method, which is modifying internal state of object. 
You can get non-const copy of foundation in a bit elegant way: change `const Mesh& foundation` argument to just `Mesh foundation`. In this way, foundation argument will be passed by value, not by const ref, and you will have non-const copy.